### PR TITLE
Enable codeql workflow

### PR DESCRIPTION
**Pull request by tylerbutler**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml
Comment (tylerbutler) [inline .github/workflows/codeql-analysis.yml:14]:
```suggestion
```

File: .github/workflows/codeql-analysis.yml
Comment (tylerbutler) [inline .github/workflows/codeql-analysis.yml:8]:
```suggestion
```

File: .github/workflows/codeql-analysis.yml
Comment (tylerbutler) [inline .github/workflows/codeql-analysis.yml:28]:
```suggestion
        language: [ 'javascript', 'typescript' ]
```

File: .github/workflows/codeql-analysis.yml
Comment (tylerbutler) [inline .github/workflows/codeql-analysis.yml:30]:
```suggestion
        # Learn more about CodeQL language support at https://git.io/codeql-language-support
```

**Review by ChumpChief — APPROVED** [review]

**Comment by anthony-murphy** [comment]
i'm a bit worried about this, as it runs really late, aka only in the pr, so problems could be costly to fix as they are not found in the dev inner loop

**Comment by tylerbutler** [comment]
> i'm a bit worried about this, as it runs really late, aka only in the pr, so problems could be costly to fix as they are not found in the dev inner loop

I agree that this is not ideal. I'm not sure it's possible to run locally, but perhaps in a container. But I don't think we should block on that at the moment. These alerts should not prevent anyone from merging at this point. (It's possible I have made config mistakes here like I did with codeowners, but if so, we'll sort them out quickly.) Before we turn on any actual gates, we'll get answers for this and other questions.

**Comment by anthony-murphy** [comment]
sounds good. also, what kind of issue will this find. i don't know anything about codeql

**Comment by tylerbutler** [comment]
The links to example issues keep disappearing, likely because we don't have a baseline in main yet, so I am going to quit chasing them and merge.

**Pull request merged**
